src/Simulation/Simulators/CommonNativeSimulator/Dump.cs

```diff
@@ -96,5 +92,393 @@ public QSimDumpRegister(CommonNativeSimulator m) : base(m)
+                (
+                    truncateSmallAmplitudes
+                    ? Amplitudes
+                        .Select((amplitude, idx) => (amplitude, idx))
+                        .Where(item =>
+                            System.Math.Pow(item.amplitude.Magnitude, 2.0) >= truncationThreshold
+                        )
+                    : Amplitudes.Select((amplitude, idx) => (amplitude, idx))
+                )
+                .Select(
+                    item => (item.amplitude, BasisStateLabel(convention, item.idx))
+                )
+                .OrderBy(
+                    item => item.Item2,
+                    // If a basis state label is numeric, we want to compare
+                    // numerically rather than lexographically.
+                    convention switch {
+                        BasisStateLabelingConvention.BigEndian => ToIntComparer,
+                        BasisStateLabelingConvention.LittleEndian => ToIntComparer,
+                        _ => Comparer<string>.Default
+                    }
+                );
+
+            /// <summary>
+            ///     Using the given labeling convention, returns the label for a
+            ///     computational basis state described by its bit string as encoded
+            ///     into an integer index in the little-endian encoding.
+            /// </summary>
+            public string BasisStateLabel(
+                BasisStateLabelingConvention convention, int index
+            ) => convention switch
+                {
+                    BasisStateLabelingConvention.Bitstring =>
+                        String.Concat(
+                            System
+                                .Convert
+                                .ToString(index, 2)
+                                .PadLeft(NQubits, '0')
+                                .Reverse()
+                        ),
+                    BasisStateLabelingConvention.BigEndian =>
+                        System.Convert.ToInt64(
+                            String.Concat(
+                                System.Convert.ToString(index, 2).PadLeft(NQubits, '0').Reverse()
+                            ),
+                            fromBase: 2
+                        )
+                        .ToString(),
+                    BasisStateLabelingConvention.LittleEndian =>
+                        index.ToString(),
+                    _ => throw new ArgumentException($"Invalid basis state labeling convention {convention}.")
+                };
+
+
+            /// <summary>
+            /// Returns a string that represents the label for the given base state.
+            /// </summary>
+            public virtual string FormatBaseState(uint idx) 
+            {
+                int qubitCount = Amplitudes?.Length ?? 0;
+                int maxCharsStateId = ((1 << qubitCount) - 1).ToString().Length;
+
+                return $"∣{idx.ToString().PadLeft(maxCharsStateId, ' ')}❭";
+            }
```

Comment:
Should this be `FormatBasisState`? Also, it looks like this method doesn't use the basis state labeling method above, but is fixed to using `BasisStateLabelingConvention.LittleEndian`. I'd actually recommend removing this method entirely, since it duplicates `BasisStateLabel` but is less configurable.

Reply:
Removing.